code/datums/elements/can_shatter.dm

```diff
@@ -53,9 +53,11 @@
 
 /// Handles the actual shattering part, throwing shards of whatever is defined on the component everywhere
 /datum/element/can_shatter/proc/shatter(atom/movable/source, atom/hit_atom)
-	var/generator/scatter_gen = generator(GEN_CIRCLE, 0, 48, NORMAL_RAND)
-	var/scatter_turf = get_turf(hit_atom)
+	var/turf/scatter_turf = get_turf(hit_atom)
+	if(!TURF_SHARES(scatter_turf)) //turf blocked(e.g by a wall, airlock, window etc i.e. anything that blocks air) so drop on the source location
```

Comment:
don't we want to check pass rules here rather then atmos? or would that break it in some other way

Reply:
The idea is anything that doesn't allow atmos to pass through is dense enough to stop shattered pieces from embedding inside it, we're talking closed doors  & airlocks, walls, grills and so on.

While my initial idea was to just to type check against a list of atoms we don't want to embed in this would full proof it against a lot of edge cases we may not have considered.

If it does cause weird behavior we can create a white list to allow embedding but I haven't found any problems so far

Reply:
could we literally use can pass or does that fuck with some things you DO want it to shatter into

Reply:
I'll test out CanPass(). That maybe sufficient, give me a minute

Reply:
Ok it doesn't work.

CanPass() is too strict for e.g. i tried throwing a plate on the table & the plate broke on the turf before it when ideally it should have shattered & scattered its pieces on the table top. 

Same thing for crates, it won't allow the pieces to fall on top of the crate when logically it should.

`CanPass()` is way too strict

Reply:
So it works and i went with it.

closed windoors don't block the thrown object however which is a bug & the shattered pieces land on the other side of it but that's a bug for another day

Reply:
is that just a closed windoor and thrown objects issue

Reply:
It's a issue with whatever uses `/obj/machinery/door/window/CanAllowThrough()` in general

Reply:
hhhhhhhhhh